clients/src/main/java/org/apache/kafka/clients/admin/DescribeTopicsOptions.java

```diff
@@ -42,6 +43,15 @@ public DescribeTopicsOptions timeoutMs(Integer timeoutMs) {
         return this;
     }
 
+    public DescribeTopicsOptions alwaysReadFromController(boolean alwaysReadFromController) {
```

Comment:
DescribeTopicsOptions is a public API so if we do want to add more interfaces we need to do a small KIP.. 

On the other hand, I'm wondering how certain we are that the root cause of the flaky test failure is due to not reading from the controller? Eventually the metadata should be propagated to other nodes as well and our default `waitUntilTrue` is taking 15 secs, so not sure if we should fix the test if it just taking more than 15 secs by changing the behavior of the admin client to always talk to controller (since it would also introduce large amount of loads on the controller).

Reply:
Regardless of whether failing reading from controller is the root cause, do you think it makes sense to offer an alternative option for users to let them select which nodes to read the metadata?

Reply:
I agree that the current metadata propagation staleness could be an issue, in fact one of the major motivations of KIP-500 is to make the metadata propagation more controllable and up-to-date.

However, I think let users select if they should read from controller or not could be a risky option, since once opens up this opportunity I'd predict everyone would select to read from controller to get most up-to-date metadata which would bring a large load on the controller.

Reply:
So, what about retrofitting `numPartitions` to accept an expected partition number and having `getTopicMetadata` more time to await?

Reply:
That sounds promising. Could you provide a revision of the PR so I can take a closer look at the proposal?